packages/kit/src/runtime/client/client.js

```diff
@@ -1190,7 +1193,7 @@ export function create_client(app, target) {
 			});
 		}
 
-		if (__SVELTEKIT_DEV__) {
+		if (__SVELTEKIT_DEV__ && status !== 404) {
```

Comment:
is it just 404 that we'd want to skip the debugger for? there might be others
```suggestion
		if (__SVELTEKIT_DEV__ && (status === 400 || status > 404)) {
```

Reply:
I chose 404 exclusively because it's the "fall back to the server who might know how to deal with this route" case. The others are more likely bugs in user code - what's a use case where a 401 would result in a full page reload? Redirect to oath?

Reply:
Ah, this code's changed quite a bit since I last looked at it, so I don't know exactly what conditions must be true for the user to end up here. I'm fine to leave it as 404 for now and we can adjust later if needed